tythonco/sfdx-project-template
# Ci GitHub action validate workflow

### To Test

1. Clone this template repo and start a fake/test project

2. Connect to a DevHub org by running the creation scrips, i.e. `npm start`

3. That should run thru the auth setup process and should result in having a "prod_auth_url.txt" and "prod_auth_url.txt.enc" files

4. Go into your fake project repo on Github and:
a) Enable Actions by going to the "Actions" tab
b) Add the `AUTH_FILE_KEY` value to Github Secrets in the repo. See:
https://help.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets

5. (optional) Create a "dev" branch for your fake project.

6. Branch off to some other branch (whatever name) and create some diffs then open a pull request to `master`.
*OR* optionally, comment out line 48 in `.github/workflows/validate.yml` so the QA workflow fallsback to "prod_auth_url" reference.

7. Goto the "Actions" tab and verify that everything is running as expected. (The flow may fail due to a failed *test*, but we want to be sure that all the authentication and commands to sfdx are working.)

8. (Optional) Assuming tests pass and check-deploy succeeds, *if the target branch for PR was "dev"*, we will kick off a quick deploy to our test org using the job-id resulting from the previous step. Ensure that the action runs as expected.
